spec/cucumber/core/test/step_spec.rb

```diff
@@ -37,22 +38,32 @@ module Cucumber::Core::Test
         expect(args_spy).to eq expected_args
       end
 
-      context "when a passing action exists" do
-        it "returns a passing result" do
+      context "passing action" do
+        it "returns a passing result when the step invocation doesn't fail" do
           test_step = Step.new(text, location).with_action {}
           expect( test_step.execute ).to be_passed
         end
+
+        it "returns a passing result when the step invocation returns a PassedInvokeResult" do
+          test_step = Step.new(text, location).with_action { PassedInvokeResult.new }
+          expect( test_step.execute ).to be_passed
+        end
       end
 
-      context "when a failing action exists" do
+      context "failing action" do
         let(:exception) { StandardError.new('oops') }
 
-        it "returns a failing result" do
+        it "returns a failing result when the step invocation raises an error" do
```

Comment:
question. Can we get a step invocation which fails from not raising an error but generating one.

I.e. I try do something like `1/0` this should raise an error implicitly. If I'm correct, could we use something implicit in the spec instead to be more descriptive

Reply:
It will be exactly the same result, both `raise exception` and `1/0` will result in an exception being thrown, but since I need the exception to do the `expect( result.exception ).to eq exception`, it simpler to manually raise one.